### PR TITLE
Change signing to uuid for create widget_id

### PR DESCRIPTION
**Pull request by CleitonDeLima**
Resolves: #581 

This PR has the objective of creating a `widget_id` using a `uuid`. 
This decreases the chances of the keys overlapping when saving to the cache.

### CODE REVIEW

**Comment by CleitonDeLima** [comment]
> However, the signing does need to stay in place. UUIDv4 are not cryptographically secure and can be guessed. Whatever ID is passed to the frontend needs to be signed to make sure the ID was given to the user and not guessed.

Now I understand the use of signing :smile:  
I will make the necessary changes

File: django_select2/forms.py [hunk 1]
```diff
@@ -245,14 +249,16 @@ def build_attrs(self, base_attrs, extra_attrs=None):
...
+                " ".join(self.dependent_fields)
+            )
 
         default_attrs.update(base_attrs)
 
         attrs = super().build_attrs(default_attrs, extra_attrs=extra_attrs)
 
-        # encrypt instance Id
-        self.widget_id = signing.dumps(id(self))
+        self.uuid = str(uuid.uuid4())
```
Comment (codingjoe) [inline django_select2/forms.py:260]: This should probably set during `__init__` and not any other method, otherwise rendering the same form twice will have strange side effects.

File: django_select2/forms.py [hunk 2]
```diff
@@ -226,8 +227,11 @@ def __init__(self, attrs=None, choices=(), **kwargs):
         if dependent_fields is not None:
             self.dependent_fields = dict(dependent_fields)
         if not (self.data_view or self.data_url):
-            raise ValueError('You must ether specify "data_view" or "data_url".')
-        self.userGetValTextFuncName = kwargs.pop('userGetValTextFuncName', 'null')
+            raise ValueError(
```
Comment (codingjoe) [inline django_select2/forms.py:230]: Please drop unrelated changes.
